src/cp.js

```diff
@@ -75,7 +77,11 @@ function copyFileSync(srcFile, destFile, options) {
     fs.closeSync(fdr);
     fs.closeSync(fdw);
 
-    fs.chmodSync(destFile, common.statFollowLinks(srcFile).mode);
+    if (options.preserve) {
+      fs.chownSync(destFile, srcStat.uid, srcStat.gid);
+      fs.utimesSync(destFile, srcStat.atime, srcStat.mtime);
+    }
+    fs.chmodSync(destFile, srcStat.mode);
```

Comment:
Instead of changing the mode after writing the file, we can set the mode on the initial call to `fs.openSync` on line 65:
```
fdw = fs.openSync(destFile, 'w', srcStat.mode);
```